lib/Common/src/IJsonConfig.h

```diff
@@ -0,0 +1,109 @@
+/* MIT License
+ *
+ * Copyright (c) 2024 - 2024 Andreas Merkle <web@blue-andi.de>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  Json Configuration interface
+ * @author Andreas Merkle <web@blue-andi.de>
+ * 
+ * @addtogroup hal
+ *
+ * @{
+ */
+
+#ifndef IJSON_CONFIG_H
+#define IJSON_CONFIG_H
+
+/******************************************************************************
+ * Compile Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include <ArduinoJson.h>
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+
+/**
+ * This interface defines the functions required for JSON bases configuration
+ * updates. It is used to provide and apply configuration settings to and from
+ * a REST API.
+ */
+class IJsonConfig
```

Comment:
Not sure whether the interface class is necessary, see my comment in PluginWithConfig.hpp
mergeConfiguration() is only required between plugin main class and its view. The other two methods are in general required for configuration file handling in PluginWithConfig.

Additional it increases the dependency from plugin library to common library. But could live in plugin library.